yt_dlp/extractor/yandexvideo.py

```diff
@@ -268,35 +270,74 @@ class ZenYandexChannelIE(InfoExtractor):
-            if key.startswith('__serverState__'):
-                data_json = data_json[key]
-        items = list(try_get(data_json, lambda x: x['feed']['items'], dict).values())
-        more = try_get(data_json, lambda x: x['links']['more']) or None
+    def _entries(self, channel_title, server_state_json, server_settings_json):
+        items = list((try_get(server_state_json, lambda x: x['feed']['items'], dict) or {}).values())
+
+        if(len(items) == 0):
+            items = list(try_get(server_settings_json, lambda x: x['exportData']['items']))
+
+        prev_next_page_id = None
+        more = try_get(
+            server_state_json, lambda x: x['links']['more']
+        ) or try_get(
+            server_settings_json, lambda x: x['exportData']['more']['link']) or None
+
+        if more:
+            next_page_id = try_get(parse_qs(more), lambda x: x['next_page_id'][0]) or None
+        else:
+            next_page_id = None
+
         for page in itertools.count(1):
-            for item in items:
+            for item in filter(lambda x: x.get('type') == 'gif', items):
                 video_id = item.get('publication_id') or item.get('publicationId')
                 video_url = item.get('link')
                 yield self.url_result(video_url, ie=ZenYandexIE.ie_key(), video_id=video_id.split(':')[-1])
-            if not more:
+
+            if any([not more, len(items) == 0, next_page_id and next_page_id == prev_next_page_id, not next_page_id]):
                 break
-            data_json = self._download_json(more, id, note='Downloading Page %d' % page)
+
+            data_json = self._download_json(more, channel_title, note='Downloading Page %d' % page)
             items = data_json.get('items', [])
             more = try_get(data_json, lambda x: x['more']['link']) or None
 
+            prev_next_page_id = next_page_id
+            if more:
+                next_page_id = try_get(parse_qs(more), lambda x: x['next_page_id'][0]) or None
+            else:
+                next_page_id = None
+
     def _real_extract(self, url):
         id = self._match_id(url)
-        return self.playlist_result(self._entries(id, url), playlist_id=id)
+        webpage = self._download_webpage(url, id)
+        data_json = self._parse_json(re.findall(r'var\s?data\s?=\s?({.+?})\s?;', webpage)[-1], id)
+        server_state_json = None
+        server_settings_json = None
+        for key in data_json.keys():
+            if key.startswith('__serverState__'):
+                server_state_json = data_json[key]
+            if key.startswith('__serverSettings__'):
+                server_settings_json = data_json[key]
+            if server_state_json and server_settings_json:
+                break
+
+        channel_title = try_get(server_state_json, lambda x: x['channel']['source']['title']) or id
+        channel_description = try_get(server_state_json, lambda x: x['channel']['source']['description']) or None
```

Comment:
You could use traverse_obj here as well. But this is fine too

Reply:
I maintained the channel_title variable (it is used twice) but used traverse_obj instead.